{

	"server": [

		{
			"action": "HTTPS Protocol Error", 
			"provider": "application",
			"number": 2001, 
			"msg": "HTTPS Server exiten on error:"
		},

		{
			"action": "HTTP Protocol Error", 
			"provider": "application",
			"number": 2002, 
			"msg": "HTTP Server exiten on error:"
		}

	],

	"uncaught": [

		{
			"action": "Generic", 
			"provider": "application",
			"number": 9000, 
			"msg": "Uncaught Exception Error: ",
			"stack": null
		}	


	],

	"critical": [

		{
			"action": "bootstrap-sequence", 
			"provider": "application",
			"number": 10000, 
			"msg": "Bootstrap sequence failed on critical error",
			"stack": null
		}

	],

	"user": [

		{
			"action": "generic-user-error", 
			"provider": "User Opertion Service",
			"number": 11000, 
			"msg": null
		},
				
		{
			"action": "authentication", 
			"provider": "google",
			"number": 11030, 
			"msg": "Authenticated Google user is not a natural person (business or organization)."
		},

		{
			"action": "authentication", 
			"provider": "google",
			"number": 11031, 
			"msg": "Authenticated Google user has no public ID or default Google Identifier."
		},

		{
			"action": "authentication", 
			"provider": "google",
			"number": 11032, 
			"msg": "No valid user name could be constructed for authenticated Google user."
		},

		{
			"action": "authentication", 
			"provider": "google",
			"number": 11033, 
			"msg": "'No email account could be extracted from authenticated Google user's profile."
		},

		{
			"action": "authentication", 
			"provider": "google",
			"number": 11034, 
			"msg": "Authenticated Google user has no public ID or default Google Identifier."
		},

		{
			"action": "query_user", 
			"provider": "User Operations",
			"number": 11100, 
			"msg": "Error occured while querying user collections for user."
		},

		{
			"action": "test_user_email", 
			"provider": "User Operations",
			"number": 11101, 
			"msg": "User has an invalid email address."
		},

		{
			"action": "test_user_password", 
			"provider": "User Operations",
			"number": 11102, 
			"msg": "User has entered an invalid password."
		},

		{
			"action": "add_user_authentication_moment", 
			"provider": "User Operations",
			"number": 11110, 
			"msg": "Could not generate authentication object aftert user logged on"
		},

		{
			"action": "test_user_firstname", 
			"provider": "User Operations",
			"number": 11120, 
			"msg": "New user has entered an invalid first name."
		},

		{
			"action": "test_user_lastname", 
			"provider": "User Operations",
			"number": 11121, 
			"message": "New user has entered an invalid last name."
		},

		{
			"action": "decrypt_user_password", 
			"provider": "User Operations",
			"number": 11130, 
			"msg": "Decryption attempt of user password with Crypto and initialization vector failed."
		},

		{
			"action": "decrypt_user_password", 
			"provider": "User Operations",
			"number": 11131, 
			"msg": "Decryption attempt of user password with default Crypto failed."
		},

		{
			"action": "decrypt_user_password", 
			"provider": "User Operations",
			"number": 11132, 
			"msg": "Decryption attempt of user password with BCrypt failed."
		},

		{
			"action": "encrypt_user_password", 
			"provider": "User Operations",
			"number": 11135, 
			"msg": "Encryption attempt of user password with Crypto and initialization vector failed."
		},

		{
			"action": "encrypt_user_password", 
			"provider": "User Operations",
			"number": 11136, 
			"msg": "Encryption attempt of user password with default Crypto failed."
		},

		{
			"action": "encrypt_user_password", 
			"provider": "User Operations",
			"number": 11137, 
			"msg": "Encryption attempt of user password with BCrypt failed."
		},


		{
			"action": "create_user_directory", 
			"provider": "User Operations",
			"number": 11150, 
			"msg": "Creation of public directory for default user failed."
		},

		{
			"action": "create_user_directory", 
			"provider": "User Operations",
			"number": 11151, 
			"msg": "Creation of public directory for customer user type failed."
		},

		{
			"action": "find_user", 
			"provider": "User Operations",
			"number": 11160, 
			"msg": "An error occured when trying to find local user, client, or customer"
		},

		{
			"action": "insert_user", 
			"provider": "User Operations",
			"number": 11170, 
			"msg": "An error occured when trying to insert a new user into the <local> database. User may already exist."
		},

		{
			"action": "insert_user", 
			"provider": "User Operations",
			"number": 11171, 
			"msg": "An error occured when trying to insert a new user into the <remote> database."
		},

		{
			"action": "update_user", 
			"provider": "User Operations",
			"number": 11180, 
			"msg": "An error occured when trying to update existing user in <local> database."
		},

		{
			"action": "update_user", 
			"provider": "User Operations",
			"number": 11181, 
			"msg": "An error occured when trying to update existing user in <remote> database."
		},


		{
			"action": "delete_user", 
			"provider": "User Operations",
			"number": 11185, 
			"msg": "An error occured when trying to remove an existing user from the database. Did you provide a field value that identifies the user? "
		},

		{
			"action": "delete_user", 
			"provider": "User Operations",
			"number": 11186, 
			"msg": "An error occured when trying to remove an existing user from the database: database model for this user type could not be allocated.? "
		},

		{
			"action": "decrypt_user_password", 
			"provider": "User Operations",
			"number": 11200, 
			"msg": "Decryption Error: could not decrypt password type 1 - Crypto with initializationVector. Did you provide an old or wrong password?"
		},

		{
			"action": "decrypt_user_password", 
			"provider": "User Operations",
			"number": 11201, 
			"msg": "Decryption Error: could not decrypt password type 2 - Default Crypto. Did you provide an old or wrong password?"
		},

		{
			"action": "decrypt_user_password", 
			"provider": "User Operations",
			"number": 11202, 
			"msg": "Decryption Error: could not decrypt password type 3 - BCrypt. Did you provide an old or wrong password?"
		},


		{
			"action": "user_password_change", 
			"provider": "User Operations",
			"number": 11203, 
			"msg": "Data Error: Old password equals new password. No change needed."
		},


		{
			"action": "user_password_change", 
			"provider": "User Operations",
			"number": 11204, 
			"msg": "Data Error: user incorrectly confirmed new password."
		},

		{
			"action": "user_password_change", 
			"provider": "User Operations",
			"number": 11205, 
			"msg": "Data Error: new password does no match defined <Password policy>"
		},

		{
			"action": "user_password_change", 
			"provider": "User Operations",
			"number": 11206, 
			"msg": "Data Error: No matching user found for this password change. Are you using the right GraphhQL mutation query?"
		},

		{
			"action": "user_password_change", 
			"provider": "User Operations",
			"number": 11207, 
			"msg": "Data Error: Password change could not be added to security history "
		},

		{
			"action": "user_address_change", 
			"provider": "User Operations",
			"number": 11300, 
			"msg": "Data Error: Invalid Form Data"
		},

		{
			"action": "user_address_change", 
			"provider": "User Operations",
			"number": 11301, 
			"msg": "Data Error: Invalid Person Database Identifier"
		},

		{
			"action": "user_address_change", 
			"provider": "User Operations",
			"number": 11302, 
			"msg": "Data Error: Invalid Person Email Identifier"
		},


		{
			"action": "user_address_change", 
			"provider": "User Operations",
			"number": 11303, 
			"msg": "Data Error: Could not execute mutation on non existing Person."
		},

		{
			"action": "user_address_change", 
			"provider": "User Operations",
			"number": 11304, 
			"msg": "Data Error: Could not identify Owner of address."
		},


		{
			"action": "user_address_change", 
			"provider": "User Operations",
			"number": 11305, 
			"msg": "Data Error: Could not find address."
		}
		


	],

	"systemuser": [

		{
			"action": "System User Service Uncaugth Error", 
			"provider": "System User Service",
			"number": 12000, 
			"msg": "System user error: "
		},

		{
			"action": "System User Profile formatting", 
			"provider": "System User Service",
			"number": 12010, 
			"msg": "System user error: "
		}

	],

	"database": [
		
		{
			"action": "save", 
			"number": 11030, 
			"msg": "An error occured when saving new user"
		}

	]
				
}